clients/src/main/java/org/apache/kafka/common/utils/MappedByteBuffers.java

```diff
@@ -67,7 +73,7 @@ public static void unmap(String resourceDescription, MappedByteBuffer buffer) th
         if (!buffer.isDirect())
             throw new IllegalArgumentException("Unmapping only works with direct buffers");
         if (UNMAP == null)
-            throw new UnsupportedOperationException(UNMAP_NOT_SUPPORTED_REASON);
+            throw UNMAP_NOT_SUPPORTED_EXCEPTION;
```

Comment:
I thought the reason the code was creating a new exception each time earlier with the saved reason was to provide a stack trace that reflected the actual call. Not sure if it matters here, but we could log the error from the static initializer and throw a new exception here with the reason if the stack trace to the call is useful.

Reply:
I considered that, but I don't think the stacktrace for the actual call matters. The original stacktrace provides more information when diagnosing issues.